### PR TITLE
wip: batch portal

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Simple huff contract for the batch portal. It is `ownable` and allows us to later upgrade to an offchain leader election mechanism.
Tests don't build yet because of a missing dep

### CODE REVIEW

File: packages/contracts-bedrock/contracts/L1/BatchPortal.huff [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  caller                      // [from, newOwner]
+  __EVENT_HASH(OwnerUpdated)  // [sig, from, newOwner]
+  0x00 0x00                   // [0, 32, sig, from, newOwner]
+  log3                        // []
+
+  stop
+}
+
+#define macro FALLBACK() = takes (0) returns (0) {
+  IS_OWNER()
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/L1/BatchPortal.huff:46]: So the sequencer is the owner? But the owner account can also call `transferOwner`, so in the event of the sequencer key being stolen, this won't help us much. There should be a separate account and auth (ie. `IS_ADMIN`) which would probably be held by the multisig.

Comment (tynes) [reply packages/contracts-bedrock/contracts/L1/BatchPortal.huff:46]: Good call

**Comment by tynes** [comment]
I think we want to check the gas cost diff between this impl and a yul based impl

**Comment by mslipper** [comment]
Closing this since we've since moved on to a different architecture.

**Pull request closed without merging**